src/network/networkctl.c

```diff
@@ -2919,6 +2930,477 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
+
+                        path = path_join(NETWORK_DIRS[0], *name);
+                        if (!path)
+                                return log_oom();
+
+                        r = edit_files_add(&context, path, NULL, NULL);
+                        if (r < 0)
+                                return r;
+                        continue;
+                }
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get the path of network config '%s': %m", *name);
+
+                r = add_config_to_edit(&context, path, dropins);
+                if (r < 0)
+                        return r;
+        }
+
+        r = do_edit_files_and_install(&context);
+        if (r < 0)
+                return r;
+
+        if (arg_no_reload)
+                return 0;
+
+        if (!sd_booted() || running_in_chroot() > 0) {
+                log_debug("System is not booted with systemd or is running in chroot, skipping reload.");
+                return 0;
+        }
+
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+
+        r = sd_bus_open_system(&bus);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to system bus: %m");
+
+        if (udevd_trigger) {
+                r = udevd_reload(bus);
+                if (r < 0)
+                        return r;
+                if (r > 0) {
+                        r = safe_fork("(udevadm)",
+                                      FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG|FORK_LOG|FORK_WAIT|FORK_FLUSH_STDIO,
+                                      NULL);
+                        if (r < 0)
+                                return r;
+                        if (r == 0) { /* Child */
+                                execlp("udevadm", "udevadm", "trigger", "--subsystem-match=net", "--action=add",
+                                       "--settle", NULL);
+
+                                log_error_errno(errno, "Failed to execute udevadm: %m");
+                                _exit(EXIT_FAILURE);
+                        }
+                }
+        }
+
+        if (networkd_reload) {
+                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+
+                if (!networkd_is_running()) {
+                        log_debug("systemd-networkd is not running, skipping reload.");
+                        return 0;
+                }
+
+                r = bus_call_method(bus, bus_network_mgr, "Reload", &error, NULL, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to reload systemd-networkd: %s", bus_error_message(&error, r));
+        }
+
+        return 0;
+}
+
+static int verb_cat(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_netlink_unrefp) sd_netlink *rtnl = NULL;
+        int r, ret = 0, has_networkd = -1;
+
+        pager_open(arg_pager_flags);
+
+        STRV_FOREACH(name, strv_skip(argv, 1)) {
```

Comment:
The loop is almost identical to the one in `verb_edit()`. Is it possible to extract the logic?